### PR TITLE
Start an Asahi version

### PR DESCRIPTION
**Pull request by ericcurtin**
Asahi has a forked version of mesa while it upstream.

### CODE REVIEW

File: container-images/ramalama/Containerfile [hunk 1]
```diff
@@ -4,8 +4,6 @@ FROM registry.access.redhat.com/ubi9/ubi:9.4-1214.1729773476
 ARG HUGGINGFACE_HUB_VERSION=0.26.2
```
Comment (rhatdan) [inline container-images/ramalama/Containerfile:4]: We don't need this anymore.

File: container-images/asahi/Containerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+FROM fedora:41
+
+# renovate: datasource=github-releases depName=huggingface/huggingface_hub extractVersion=^v(?<version>.*)
+ARG HUGGINGFACE_HUB_VERSION=0.26.2
```
Comment (rhatdan) [inline container-images/asahi/Containerfile:4]: Don't need this anymore.

File: container-images/cuda/Containerfile [hunk 3]
```diff
@@ -32,8 +30,7 @@ RUN dnf install -y python3 python3-pip && \
     rm -rf /var/cache/*dnf*
 
 # Install Python packages in the runtime image
-RUN pip install "huggingface_hub==${HUGGINGFACE_HUB_VERSION}" \
-      "omlmd==${OMLMD_VERSION}"
+RUN pip install "omlmd==${OMLMD_VERSION}"
```
Comment (rhatdan) [inline container-images/cuda/Containerfile:33]: Don't need this either.

File: container-images/ramalama/Containerfile [hunk 4]
```diff
@@ -24,9 +20,7 @@ RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.n
     rm -rf /var/cache/*dnf*
 
 RUN /usr/bin/python3 --version
-RUN pip install "huggingface_hub==${HUGGINGFACE_HUB_VERSION}"
 RUN pip install "omlmd==${OMLMD_VERSION}"
```
Comment (rhatdan) [inline container-images/ramalama/Containerfile:23]: Don't need this.

**Comment by rhatdan** [comment]
Does this mean on Asahi we can't use the standard RamaLama image?

**Comment by ericcurtin** [comment]
Fedora Asahi Remix is 99% percent Fedora... The two main packages it forks are the kernel and mesa (mesa being where the userspace GPU driver is).

This Vulkan driver "Honeykrips" is weeks old, so the patches have not made their way to mainline Fedora mesa yet:

https://www.phoronix.com/news/Mesa-HoneyKrisp-October

anyway, with the other images being UBI based, even besides this mesa fork, we would have had to make a special container image for Asahi because Asahi is only enabled for Fedora, not RHEL.

So yeah, we need a specific container image for Asahi.

I don't expect to make this exception often, but the Asahi community is so vibrant and kinda is a flagbearer for ARM Linux efforts and ARM GPU drivers. I think it's worth it.

**Comment by ericcurtin** [comment]
@conan-kudo expressed an interesting in running llama.cpp outside of containers which is fine too, as long as you are using the forked mesa and building/using llama.cpp built with -DGGML_VULKAN=1 it should work

**Comment by rhatdan** [comment]
We should be able to build the asahi image from the RamaLama Containerfile.

podman build --from fedora:41 -t quay.io/ramalama/asahi:latest containers-images/ramalama

That way we don't need to update multiple containerfiles.

**Comment by ericcurtin** [comment]
> We should be able to build the asahi image from the RamaLama Containerfile.
> 
> podman build --from fedora:41 -t quay.io/ramalama/asahi:latest containers-images/ramalama
> 
> That way we don't need to update multiple containerfiles.

Other than:

**Comment by rhatdan** [comment]
No, I was thinking they were the same Containerfiles, other then the From, I will merge and we can revisit later.

**Pull request merged**